backend/models/users.py

```diff
@@ -27,3 +32,27 @@ class User(db.Model):
     delete_date = db.Column(db.TIMESTAMP)
 
     role = db.relationship(u'Role')
+
+    def is_deleted(self):
+        return bool(self.delete_date)
+
+    def is_last_admin(self):
+        admins = User.query.filter_by(role_id=User.ROLE_ADMIN, delete_date=None)
+        count = admins.count()
+        if count > 1:
+            return False
+        return True
+
+    def delete(self):
+        if self.role_id == User.ROLE_ADMIN:
+            if not self.is_last_admin():
+                self.delete_date = func.current_timestamp()
+                db.session.commit()
```

Comment:
It's a bad idea to make commit here, please move it controller. In future you may want to add some other model methods, and they will cause unneeded transactions on each change.